caom2/src/main/java/ca/nrc/cadc/caom2/types/MultiPolygon.java

```diff
@@ -0,0 +1,367 @@
+*
+*  $Revision: 5 $
+*
+************************************************************************
+*/
+
+package ca.nrc.cadc.caom2.types;
+
+import ca.nrc.cadc.util.HexUtil;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ *
+ * @author pdowler
+ */
+public class MultiPolygon implements Serializable
+{
+    private static final long serialVersionUID = 201708141600L;
+    
+    private final List<Vertex> vertices = new ArrayList<Vertex>();
+
+    // lazily computed
+    private transient Point center;
+    private transient Double area;
+    private transient Circle minimumSpanningCircle;
+    private transient Boolean ccw;
+
+    public MultiPolygon() { }
+
+    private void validate()
+    {
+        if (vertices.size() < 4) // triangle
+            throw new IllegalPolygonException("invalid polygon: " + vertices.size() + " vertices");
+
+        Vertex end = vertices.get(0);
+        if ( !SegmentType.MOVE.equals(end.getType()) )
+            throw new IllegalPolygonException("invalid polygon: first vertex must be MOVE, found " + end);
+        end = vertices.get(vertices.size() - 1);
+        if ( !SegmentType.CLOSE.equals(end.getType()) )
+            throw new IllegalPolygonException("invalid polygon: last vertex must be CLOSE, found " + end);
+
+        boolean openLoop = false;
+        for (Vertex v : vertices)
+        {
+            if ( SegmentType.MOVE.equals(v.getType()) )
+            {
+                if (openLoop)
+                    throw new IllegalPolygonException("invalid polygon: found MOVE when loop already open");
+                openLoop = true;
+            }
+            else if (  SegmentType.CLOSE.equals(v.getType()) )
+            {
+                if (!openLoop)
+                    throw new IllegalPolygonException("invalid polygon: found CLOSE without MOVE");
+                openLoop = false;
+            }
+        }
+    }
+
+    public List<Vertex> getVertices()
+    {
+        return vertices;
+    }
+
+    public boolean getCCW()
+    {
+        if (ccw == null)
+            initProps();
+        return ccw;
+    }
+    
+    
+    public double getArea()
+    {
+        if (area == null)
+            initProps();
+        return area;
+    }
+
+    public Point getCenter()
+    {
+        if (center == null)
+            initProps();
+        return center;
+    }
+
+    /**
+     * @return 
+     */
+    public double getSize()
+    {
+        if (minimumSpanningCircle == null)
+            initProps();
+        return minimumSpanningCircle.getSize();
+    }
+
+    public Circle getMinimumSpanningCircle()
+    {
+        if (minimumSpanningCircle == null)
+            initProps();
+        return minimumSpanningCircle;
+    }
+
+    public boolean isSimple()
+    {
+        validate();
+        int num = 0;
+        for (Vertex v : vertices)
+        {
+            if ( SegmentType.MOVE.equals(v.getType()) )
+                num++;
```

Comment:
No point in looping through all the vertices when it can exit upon finding the second MOVE.